pennylane/qnode.py

```diff
@@ -616,6 +619,40 @@ def __call__(self, *args, **kwargs):
         )
         self._tape_cached = using_custom_cache and self.tape.hash in cache
 
+        if qml.query_return():
+            res = qml.execute_new(
+                [self.tape],
+                device=self.device,
+                gradient_fn=self.gradient_fn,
+                interface=self.interface,
+                gradient_kwargs=self.gradient_kwargs,
+                override_shots=override_shots,
+                **self.execute_kwargs,
+            )
+
+            if not isinstance(self._qfunc_output, tuple) and not isinstance(
+                self._qfunc_output, qml.measurements.MeasurementProcess
+            ):
+                if not self.device._shot_vector:
+                    res = type(self.tape._qfunc_output)(res[0])
+                else:
+                    res = [type(self.tape._qfunc_output)(r) for r in res[0]]
+                    res = tuple(res)
+            else:
+                # Autograd or tensorflow: they do not support tuple return with backpropagation
+                backprop = False
+                if not isinstance(self._qfunc_output, qml.measurements.MeasurementProcess):
+                    backprop = any(qml.math.in_backprop(x) for x in res[0])
+                if (
+                    self.interface in ("tf", "autograd")
+                    and self.gradient_fn == "backprop"
+                    and backprop
+                ):

```

Comment:
Do we then go with the decision that we'll do the required processing here to go from a tuple to array (instead of requiring the user to do it)?

Reply:
Yes I think it is  best decision because it is also impacting Tensorflow. I will add it to the ADR.